internal/docstore/mongodocstore/codec.go

```diff
@@ -118,8 +137,14 @@ func (d decoder) AsString() (string, bool) {
 }
 
 func (d decoder) AsInt() (int64, bool) {
-	i, ok := d.val.(int64)
-	return i, ok
+	switch v := d.val.(type) {
+	case int64:
+		return v, true
+	case int32:
+		return int64(v), true
```

Comment:
Can it be `int`? How about

```go
case int64:
case int32:
case int:
  return int64(v), true
```

Reply:
I don't think it can: http://bsonspec.org/spec.html.